include/aspect/introspection.h

```diff
@@ -538,14 +550,14 @@ namespace aspect
        * Get the indices of the compositional fields which are of a
        * particular type (chemical composition, porosity, etc.).
        */
-      const std::vector<unsigned int>
+      const std::vector<unsigned int> &
       get_indices_for_fields_of_type (const CompositionalFieldDescription::Type &type) const;
 
       /**
        * Get the names of the compositional fields which are of a
        * particular type (chemical composition, porosity, etc.).
        */
-      const std::vector<std::string>
+      const std::vector<std::string> &
```

Comment:
That was a bug to begin with. The fact that we marked these return types as `const` suggests that we always intended for them to be references.